subprojects/docs/src/docs/userguide/jvm/toolchains.adoc

```diff
@@ -93,6 +93,21 @@ The Java major version, the vendor (if specified) and implementation (if specifi
 will be tracked as an input for compilation and test execution.
 ====
 
+=== Configuring toolchain specifications
```

Comment:
I think the deprecation should go to `upgrading_version_7.adoc`.

And the documentation about validity could be a Javadoc on `ToolchainSpec`. I am not sure if it makes sense to add the validity here. Maybe it should be more a remark that you need to request at least a Java version.

Reply:
There should be a note in the upgrading guide indeed, but I think the mention in this page makes sense.

Reply:
I also think that we should keep this section in the guide, to have the contract in the prose. Adding the same information to the `JavaToolchainSpec` is not gonna hurt.

Reply:
Done